packages/tools/pie-css/scss/mixins/_focus.scss

```diff
@@ -0,0 +1,17 @@
+// ----------------------------------------------
+// Mixin for adding focus styles.
+// The `focus` mixin allows you to apply a box-shadow with inner and outer focus styles to an element.
+// -
+// @example
+// &:focus-visible {
+//   @include focus;
+// }
+// -
+// Outputs:
+// &:focus-visible {
+//   box-shadow: 0 0 0 2px var(--dt-color-focus-inner), 0 0 0 4px var(--dt-color-focus-outer);
+// }
+// ----------------------------------------------
+@mixin focus() {
+    box-shadow: 0 0 0 2px var(--dt-color-focus-inner), 0 0 0 4px var(--dt-color-focus-outer);
```

Comment:
I think we should add `outline: none;` as well. @RLumleyJET flagged that our shadow is conflicting with the browser's native focus styles in multiple component design reviews. 

I added that to my link pr https://github.com/justeattakeaway/pie/pull/834 but it makes sense to add it to the mixin directly.